libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c

```diff
@@ -761,74 +870,76 @@ static MQTTStatus_t handleIncomingPublish( MQTTPacketInfo_t * packet,
 
     if( status == MQTTSuccess )
     {
-        status = transportSerializePublish( pBuffer, &publishInfo, packetIdentifier );
+        status = transportSerializePublish( streamBuffer, &publishInfo, packetIdentifier );
     }
 
     return status;
 }
 
 
-static MQTTStatus_t handleIncomingSuback( MQTTPacketInfo_t * packet,
-                                          MQTTFixedBuffer_t * pBuffer )
+static MQTTStatus_t handleIncomingSuback( StreamBufferHandle_t streamBuffer,
+                                          MQTTPacketInfo_t * packet )
 {
     MQTTStatus_t status = MQTTSuccess;
     uint16_t packetIdentifier = 0;
+    uint8_t buffer[ SIZE_OF_SUB_ACK ] = { 0 };
 
     LogDebug( ( "Processing incoming SUBACK from channel." ) );
     status = IotBleMqtt_DeserializeSuback( packet, &packetIdentifier );
 
     if( status == MQTTSuccess )
     {
-        status = transportSerializeSuback( pBuffer, packet->type, packetIdentifier );
-    }
+        buffer[ 0 ] = packet->type;
+        buffer[ 1 ] = 3;
+        buffer[ 2 ] = ( uint8_t ) ( packetIdentifier >> 8 );
+        buffer[ 3 ] = ( uint8_t ) ( packetIdentifier & 0x00FFU );
+        buffer[ 4 ] = 1;
 
-    if( status == MQTTSuccess )
-    {
-        ( void ) xStreamBufferSend( streamBuffer, pBuffer->pBuffer, SIZE_OF_SUB_ACK, pdMS_TO_TICKS( RECV_TIMEOUT_MS ) );
+        ( void ) xStreamBufferSend( streamBuffer, buffer, SIZE_OF_SUB_ACK, pdMS_TO_TICKS( RECV_TIMEOUT_MS ) );
```

Comment:
SIZE_OF_SUB_ACK may not be a constant as mentioned in the above comment.

Reply:
See comment https://github.com/aws/amazon-freertos/pull/2535#discussion_r502518660

Reply:
Yea. Thanks